src/coreclr/src/System.Private.CoreLib/src/System/Reflection/RuntimeAssembly.cs

```diff
@@ -309,30 +309,15 @@ public override IList<CustomAttributeData> GetCustomAttributesData()
             return CustomAttributeData.GetCustomAttributesInternal(this);
         }
 
-        internal static RuntimeAssembly InternalLoad(string assemblyString, ref StackCrawlMark stackMark, AssemblyLoadContext? assemblyLoadContext = null)
-        {
-            AssemblyName an = new AssemblyName(assemblyString);
-
-            return InternalLoadAssemblyName(an, ref stackMark, assemblyLoadContext);
-        }
+        internal static RuntimeAssembly InternalLoad(string assemblyName, ref StackCrawlMark stackMark, AssemblyLoadContext? assemblyLoadContext = null)
+            => InternalLoad(new AssemblyName(assemblyName), ref stackMark, assemblyLoadContext);
 
-        internal static RuntimeAssembly InternalLoadAssemblyName(AssemblyName assemblyRef, ref StackCrawlMark stackMark, AssemblyLoadContext? assemblyLoadContext = null)
-        {
-            assemblyRef = (AssemblyName)assemblyRef.Clone();
-            if (assemblyRef.ProcessorArchitecture != ProcessorArchitecture.None)
-            {
-                // PA does not have a semantics for by-name binds for execution
```

Comment:
Deleted ProcessorArchitecture from the AssemblySpec to make sure that this is not necessary. It was not ever used for anything.